CHECK_POWER_COMMANDS/Check_Power_Cmds.py

```diff
@@ -0,0 +1,196 @@
+        rules_fired = rules.ORB_CMD_SW_rule_set(present_cmd, system_state.state, violations_list)
+        # If any rules fired append them to the rules_filred list
+        if rules_fired:
+            all_rules_fired.append(list(rules_fired))
+
+    # You will be looking at the next row
+    array_row_number += 1
+    present_cmd = system_packets[array_row_number]
+
+# So you've captured the information from the first ACISPKT so now
+# you have something against which to compare the times of the next
+# ACISPKT that you see.  
+#
+# Save it so that any number of commands that are NOT ACISPKT commands
+# that lie between this command and the next ACISPKT command do not
+# interfere with checking the command timing. For example:
+#
+# ACISPKT command
+# OORMPEN
+# ACISPKT command
+bfc.write_previous_ACISPKT_cmd(present_cmd)
+# Record the date of this command in system_state.state
+system_state.state['date_cmd'] = present_cmd['event_date']
+system_state.state['time_cmd'] = present_cmd['event_time']
+
+# And since this is going to be "previous" then set last_state 
+last_state = dict(system_state.state)
+
+#
+# Continue processing set the present packet to the NEXT one
+# for eachpacket in system_packets[array_row_number+1:125]:
+for eachpacket in system_packets[array_row_number+1:]:
+    rules_fired = []
+    # Record the date of this command in system_state.state
+    system_state.state['date_cmd'] = eachpacket['event_date']
+    system_state.state['time_cmd'] = eachpacket['event_time']
+
+    if eachpacket['cmd_type'] == 'ACISPKT':
+        # Save the present system state in another variable
+        last_state = dict(system_state.state)
+        # Run the ACISPKT ruleset once
+        new_rules_fired, violations_list = rules.ACISPKT_rules(eachpacket,
+                                                               last_state,
+                                                               system_state,
+                                                               bfc,
+                                                               violations_list)
+        # Append all the rules that may have fired
+        if new_rules_fired:
+            all_rules_fired.append(list(new_rules_fired))
+
+        # Keep on running the ACISPKT rules until the state does not change
+        while last_state != system_state.state:
+            # Set the last state to the present state
+            last_state = dict(system_state.state)
+            # Run the ACISPKT ruleset again
+            new_rules_fired, violations_list = rules.ACISPKT_rules(eachpacket,
+                                                                   last_state,
+                                                                   system_state,
+                                                                   bfc,
+                                                                   violations_list)
+            # Append all the rules that may have fired
+            if new_rules_fired:
```

Comment:
Same comment as above about checking the length

Reply:
implemented

Reply:
not implemented